js/util/polygon_poi.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+var Queue = require('tinyqueue');
+var Point = require('point-geometry');
+var distToSegmentSquared = require('./intersection_tests').distToSegmentSquared;
+var isClosedPolygon = require('./util').isClosedPolygon;
+
+/**
+ * Finds an approximation of a polygon's Pole Of Inaccessibiliy https://en.wikipedia.org/wiki/Pole_of_inaccessibility
+ * This is a copy of http://github.com/mapbox/polylabel adapted to use Points
+ *
+ * @param {Array<Array<Point>>} polygons first item in array is the outer ring polygon followed optionally by the set of holes, should be the first element of the result of util/classify_rings
+ * @param {number} [precision=1] Specified in input coordinate units. If 0 returns after first run, if > 0 repeatedly narrows the search space until the radius of the area searched for the best pole is less than precision
+ * @param {bool} [debug=false] Print some statistics to the console during execution
+ *
+ * @returns {Point} Pole of Inaccessibiliy.
+ */
+module.exports = function (polygons, precision, debug) {
```

Comment:
`polygons` is a bit confusing here because it really means rings of a single polygon in this case. Perhaps `rings` would be better?